model/Check/System/AdvancedSearch/AdvancedSearchIndexationCheck.php

```diff
@@ -0,0 +1,167 @@
+use oat\tao\helpers\Template;
+use oat\oatbox\reporting\Report;
+use oat\tao\model\AdvancedSearch\AdvancedSearchChecker;
+use oat\taoAdvancedSearch\model\Index\Report\IndexSummarizer;
+
+class AdvancedSearchIndexationCheck extends AbstractAdvancedSearchCheck
+{
+    public const PARAM_ALLOWABLE_PERCENTAGE = 'allowablePercentage';
+    public const PARAM_BLACKLISTED_INDEXES = 'blacklistedIndexes';
+
+    /** @var float */
+    private $allowablePercentage;
+
+    /** @var array */
+    private $blacklistedIndexes;
+
+    /** @var Report[] */
+    private $reports = [];
+
+    /** @var Report[] */
+    private $errorReports = [];
+
+    public function __construct(array $params = [])
+    {
+        $this->allowablePercentage = (float) ($params[self::PARAM_ALLOWABLE_PERCENTAGE] ?? 80);
+        $this->blacklistedIndexes = $params[self::PARAM_BLACKLISTED_INDEXES] ?? [];
+
+        parent::__construct($params);
+    }
+
+    public function getDetails(): string
+    {
+        return __('Indexes population');
+    }
+
+    protected function getTemplate() : string
+    {
+        return Template::getTemplate('Reports/advancedSearchIndexesReport.tpl', 'taoSystemStatus');
+    }
+
+    protected function doCheck(): common_report_Report
+    {
+        $advancedSearchChecker = $this->getAdvancedSearchChecker();
+
+        if (!$advancedSearchChecker->isEnabled()) {
+            return Report::createError('Advanced search disabled');
+        }
+
+        if (!$advancedSearchChecker->ping()) {
+            return Report::createError('Advanced search unavailable');
+        }
+
+        try {
+            $processedIndexesCount = 0;
+
+            foreach ($this->getIndexSummarizer()->summarize() as $data) {
+                if (
+                    !$data['totalInDb']
+                    || $data['percentageIndexed'] === 100.0
+                    || in_array($data['index'], $this->blacklistedIndexes, true)
+                ) {
+                    continue;
+                }
+
+                $this->addReport($this->createReport($data));
+                ++$processedIndexesCount;
+            }
+
+            return $this->buildFinalReport($processedIndexesCount);
+        } catch (Throwable $exception) {
+            return Report::createError('Unexpected error');
+        }
+    }
+
+    private function createReport($data): Report
+    {
+        $type = $this->getTypeByPercentage($data['percentageIndexed']);
+        $message = __(
+            'Index "%s": %d/%d (%s%%)',
+            $data['index'],
+            $data['totalIndexed'],
+            $data['totalInDb'],
+            $data['percentageIndexed']
+        );
+
+        return new Report($type, $message);
+    }
+
+    private function addReport(Report $report): void
+    {
+        $this->reports[] = $report;
+
+        if ($report->getType() === Report::TYPE_ERROR) {
+            $this->errorReports[] = $report;
+        }
+
+        $this->{'log' . $report->getType()}(
```

Comment:
I do not trust too much in this type of solution. If we add new log type here we might get a fatal error. I know in theory it is not the case now, but as a heads up - I would be more splicity in the method I am calling

Reply:
If it's ok, I will change this one to `logWarning` (for both error and warning reports)